feat(integrations/whatsapp) Send stickers via Image cards

Gordon-BP opened the pull request:
New logic to send certain images as animated/static stickers for WhatsApp.

WhatsApp imposes [strict restrictions on stickers:](https://developers.facebook.com/docs/whatsapp/cloud-api/messages/sticker-messages)
1. Only .webp format allowed
2. Smaller than 512 kb
3. Exactly 512 x 512 size

If any of the above criteria are broken by a .webp image, you get a warning in the log and the image is not sent.

![image](https://github.com/user-attachments/assets/175c7f8e-dd71-41d4-89b4-167fa3cecf14)

WhatsApp only supports .webp format for stickers, so we can't even send as a normal image as backup.

Tagging @SebastienPoitras who might also be working on features for the WhatsApp integration.

Gordon-BP committed:
new image message generator and verison bump

Gordon-BP committed:
updated hub.md, bettter error msgs

SebastienPoitras commented:
Hey Gordon,

Thank you for your contribution! We are currently working on improvements to the integration and need to merge some PRs before incorporating your changes. I'll keep this PR open for now, and we'll merge it once the ongoing work is complete.

Sébastien

SebastienPoitras committed:
Fix formatting, linting and type errors

SebastienPoitras committed:
Merge remote-tracking branch 'origin/master' into gc/integrations/whatsapp-stickers

SebastienPoitras committed:
Fix merge error

SebastienPoitras committed:
Refactor

SebastienPoitras committed:
Remove sticker images infos from hub.md

This information will be added to the WhatsApp integration guide in the
documentation instead

SebastienPoitras committed:
Refactor

SebastienPoitras commented:
Closing this PR as these changes will be merged from PR #13862 instead.

Gordon-BP commented:
Great, thanks!

Sent from [Proton Mail](https://proton.me/mail/home) for iOS

On Thu, May 15, 2025 at 23:57, Sébastien Poitras ***@***.***(mailto:On Thu, May 15, 2025 at 23:57, Sébastien Poitras <<a href=)> wrote:

> SebastienPoitras left a comment [(botpress/botpress#13839)](https://github.com/botpress/botpress/pull/13839#issuecomment-2884335399)
>
> Closing this PR as these changes will be merged from PR [#13862](https://github.com/botpress/botpress/pull/13862) instead.
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/botpress/botpress/pull/13839#issuecomment-2884335399), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/ASPXTLE3C62KTZVW5VBBLET26S2PZAVCNFSM6AAAAAB4WQ5T5OVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQOBUGMZTKMZZHE).
> You are receiving this because you authored the thread.Message ID: ***@***.***>
